tests/helm/testdata/goldenfile/logs_otc_serviceaccount/basic.output.yaml

```diff
@@ -3,6 +3,7 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: RELEASE-NAME-sumologic-otelcol-logs-collector
+  annotations: !!null
```

Comment:
it doesn't look correct

Reply:
the helm template tests do pass

Reply:
Ok, I will elaborate. I don't like how does it look and IMHO we should:

- have annotations, or
- do not `annotations` key at all, or
- have empty dictionary `{}`

Also this change is unrelated to the PR :thinking: